torch/distributed/_tensor/experimental/tp_transform.py

```diff
@@ -65,28 +72,58 @@ def __init__(
         world_size: int,
         state_dict: Dict[str, torch.Tensor],
         graph_signiture: ExportGraphSignature,
+        parallel_strategies: Dict[str, ParallelStyle],
     ) -> None:
         super().__init__()
         self.rank = rank
         self.mesh = DeviceMesh("cuda", torch.arange(world_size))
         self.state_dict: Dict[str, torch.Tensor] = state_dict
         self.graph_signiture = graph_signiture
+        self.parallel_strategies = parallel_strategies
 
     def call(self, graph_module) -> PassResult:
         gm = copy.deepcopy(graph_module)
 
-        placement_strategies = _mark_sharding(gm, self.graph_signiture, self.mesh)
+        parameter_placements = _generate_parameter_and_buffer_placements(
+            self.graph_signiture, self.parallel_strategies
+        )
+        placement_strategies = _mark_sharding(
+            gm, self.graph_signiture, self.mesh, parameter_placements
+        )
         _partitioner(gm)
         _shard_state_dict(
             self.state_dict, placement_strategies, self.graph_signiture, self.mesh
         )
         return PassResult(gm, True)
 
 
+def _generate_parameter_and_buffer_placements(
+    graph_signiture: ExportGraphSignature,
+    parallel_strategies: Dict[str, ParallelStyle],
+) -> Dict[str, Placement]:
+    """
+    Build parameter placements based on the give parallel style of linear layers.
+    """
+    parameter_placements: Dict[str, Placement] = {}
+    for linear_fqn, parallel_style in parallel_strategies.items():
+        weight_fqn = f"{linear_fqn}.weight"
+        bias_fqn = f"{linear_fqn}.bias"
```

Comment:
I think because linear does not always have a `bias` parameter, we should check if the linear_fqn passed in have bias or not?

Or we can return the placement here but later when we apply the placements to the linear related parameters, we check if there's a bias parameter at that time.